course_info/static/course_info/js/controllers/DetailsController.js

```diff
@@ -1,70 +1,115 @@
+
+        dc.init = function() {
+            var instances = courseInstances.instances;
+            if (instances && instances[dc.courseInstanceId]) {
+                dc.courseInstance = dc.getFormattedCourseInstance(
+                    instances[dc.courseInstanceId]);
+            }
+            dc.fetchCourseInstanceDetails(dc.courseInstanceId);
+        };
+
+        dc.alertPresent = function(alertType, alertKey) {
+            return (dc.alerts[alertType][alertKey] &&
+                    dc.alerts[alertType][alertKey].show);
+        };
 
-        var remove_quotes_regex = new RegExp("^\"|\"$", "g");
+        dc.handleAjaxErrorResponse = function(r) {
+            dc.handleAjaxError(
+                r.data, r.status, r.headers, r.config, r.statusText);
+        };
 
         dc.handleAjaxError = function (data, status, headers, config, statusText) {
             $log.error('Error attempting to ' + config.method + ' ' + config.url +
                 ': ' + status + ' ' + statusText + ': ' + JSON.stringify(data));
         };
 
-        dc.handleLookupResults = function(results) {
-            var courseResult = results[0];
-            var membersResult = results[1];
+        dc.isCourseInstanceEditable = function(courseRegistrarCode) {
+            // TLT-2376: sandbox and ILE courses are editable, and are
+            // identified by their course (registrar) code
+            return (courseRegistrarCode.startsWith('ILE-') ||
+                    courseRegistrarCode.startsWith('SB-'));
+        };
+
+        dc.isDefined = function(obj) {
+            return (typeof obj === 'undefined');
+        };
 
+        dc.handleCourseInstanceResponse = function(response) {
             //check if the right data was obtained before storing it
-            if (courseResult.data.course_instance_id == dc.courseInstanceId) {
-                courseInstances.instances[courseResult.data.course_instance_id] = courseResult.data;
-                dc.courseInstance = dc.getFormattedCourseInstance(courseResult.data, membersResult.data)
+            if (response.data.course_instance_id == dc.courseInstanceId) {
+                courseInstances.instances[response.data.course_instance_id] = response.data;
+                // if people/members are fetched first, we don't want to
+                // overwrite the members attribute of dc.courseInstance
+                $.extend(dc.courseInstance,
+                    dc.getFormattedCourseInstance(response.data));
+                // TLT-2376: only sandbox and ILE courses are currently editable
+                var rc = response.data.course.registrar_code;
+                dc.editable = dc.isCourseInstanceEditable(rc);
+                dc.resetForm();
             } else {
                 $log.error(' CourseInstance record mismatch for id :'
-                    + dc.courseInstanceId + ',  fetched record for :' + courseResult.data.id);
+                    + dc.courseInstanceId + ',  fetched record for :'
```

Comment:
Can we make this `', instead received record for :'`, so it's more obvious which one we asked for/were expecting, and which one we got?